fhirpipe/api/routes.py

```diff
@@ -64,7 +65,9 @@ def preview(resource_id, primary_key_value):
 
     try:
         # Connect to DB and run
-        fhir_object = fp_preview(resource_id, [primary_key_value], credentials)
+        fhir_object = fp_preview(resource_id, [primary_key_value], credentials)[0]
+    except ValidationError as e:
+        return str(e)
```

Comment:
I suggest that we return both the error and the transformed instance (even if there is no errors, we would have a graphql-style response like {data: <preview_instance>, errors: [<validation_error>]})

Reply:
I totally agree on having both the error and the preview!
But I'm not sure I can write both in fields data and errors with flask. Do you think something like `Response.data = {preview: <preview_instance>, errors: [<validation_error>]}` would be ok?

Reply:
I'm not sure I get what you mean, can't you do `return jsonify({preview: {}, errors: [<validation_errors>]}, success=True|False)` ?

Reply:
I can (without the `success` though), that's what I was saying ^^
I was just saying that on the client side, we would still have to use response.data.preview and response.data.errors, not response.data and response.errors

Reply:
oh yeah right, that's not an issue